lib/api/src/externals/global.rs

```diff
@@ -42,14 +42,9 @@ impl Global {
             ty: val.ty(),
         });
         unsafe {
-            match val {
-                Val::I32(x) => *global.get_mut().as_i32_mut() = x,
-                Val::I64(x) => *global.get_mut().as_i64_mut() = x,
-                Val::F32(x) => *global.get_mut().as_f32_mut() = x,
-                Val::F64(x) => *global.get_mut().as_f64_mut() = x,
-                Val::V128(x) => *global.get_mut().as_u128_bits_mut() = x.to_ne_bytes(),
-                _ => return Err(RuntimeError::new(format!("create_global for {:?}", val))),
-            }
+            global
+                .set(val.clone())
```

Comment:
I wish we didn't need to clone here. I guess the only reason is for the error message?

Reply:
It's a relatively small type that doesn't allocate on Clone so I'm not too worried, but yeah, the only reason for it is the error message. for what it's worth, I'd expect Rust to optimize this decently and to produce roughly the same code as was being generated before